File: src/services/template.service.ts
import api from './api';

export async function getTemplates(params?: Record<string, any>) {
  return api.get('/templates', { params });
}

export async function getTemplate(id: string) {
  return api.get(`/templates/${id}`);
}

export async function deleteTemplate(id: string) {
  return api.delete(`/templates/${id}`);
}

export async function previewTemplate(id: string) {
  return api.get(`/templates/${id}/preview`);
}

export async function applyTemplate(id: string, data: Record<string, any>) {
  return api.post(`/templates/${id}/apply`, data);
}

export async function duplicateTemplate(id: string) {
  return api.post(`/templates/${id}/duplicate`);
}

export async function exportTemplate(id: string) {
  return api.get(`/templates/${id}/export`);
}

export async function importTemplate(data: Record<string, any>) {
  return api.post('/templates/import', data);
}

File: syncflow-java/syncflow-common/src/main/java/com/syncflow/common/service/template/TemplateService.java
package com.syncflow.common.service.template;

import com.syncflow.common.dto.template.CreateTemplateDTO;
import com.syncflow.common.dto.template.TemplateVO;
import com.syncflow.common.result.PageResult;
import org.springframework.web.multipart.MultipartFile;

/**
 * Template service interface.
 */
public interface TemplateService {

    /**
     * Paginated template list.
     */
    PageResult<TemplateVO> getTemplateList(String keyword, int pageNum, int pageSize);

    /**
     * Template detail by id.
     */
    TemplateVO getTemplateDetail(Long id);

    /**
     * Create a new template.
     */
    TemplateVO createTemplate(CreateTemplateDTO dto);

    /**
     * Update an existing template.
     */
    TemplateVO updateTemplate(Long id, CreateTemplateDTO dto);

    /**
     * Delete a template (soft delete).
     */
    void deleteTemplate(Long id);

    /**
     * Preview a template.
     */
    TemplateVO previewTemplate(Long id);

    /**
     * Apply a template (increments usage count).
     */
    void applyTemplate(Long id);

    /**
     * Duplicate a template.
     */
    TemplateVO duplicateTemplate(Long id);

    /**
     * Export a template as JSON string.
     */
    String exportTemplate(Long id);

    /**
     * Import a template from JSON string.
     */
    TemplateVO importTemplate(String templateJson);
}
